src/data/list/basic.lean

```diff
@@ -1519,6 +1529,33 @@ lemma nth_le_take' (L : list α) {i j : ℕ} (hi : i < (L.take j).length) :
   nth_le (L.take j) i hi = nth_le L i (lt_of_lt_of_le hi (by simp [le_refl])) :=
 by { simp at hi, rw nth_le_take L _ hi.1 }
 
+lemma nth_take_of_lt {l : list α} {n m : ℕ} (h : m < n) :
```

Comment:
I think I would call this much more useful lemma (which basically completely characterizes `l.take n`, since it has length at most `n`) `nth_take`...